### ISSUE TITLE
Cloudformation error

### ISSUE THREAD

**Issue by christianswanepoel** [Senzing/aws-cloudformation-ecs-poc-simple#71]
CloudFormation failed to complete the install but nicely exited with "ROLLBACK_COMPLETE".

The last error event was:

"ROLLBACK_IN_PROGRESS" 
"The following resource(s) failed to create: [IamPolicySqsProducer, IamPolicyS3, IamPolicyEfs, Ec2VpcGatewayAttachment, Ec2SubnetPublic2, IamPolicyCognito, IamPolicyEc2FlowLogs, Ec2SubnetPrivate1, IamPolicyLoggingCreateStream, IamPolicySqsConsumer, IamPolicyCertificateManager, Ec2SubnetPrivate2, IamPolicyTaskRunner, IamPolicyRoute53, RdsDbClusterParameterGroup, Ec2SubnetPublic1, IamPolicyPassRole, IamPolicyCertificates, LambdaFunctionGenerateCertificate]. Rollback requested by user."

I am guessing its a permissions issue(?) - any advice?

**Comment by docktermj** [comment]
Hi @christianswanepoel !

So in a case like that, visit the Cloudformation's "Events" tab and look for the first event that failed.   The list that you show is the summary, but doesn't indicate the initial error.   Once the initial error is identified, we'll have something to work with.

**Comment by christianswanepoel** [comment]
Hey @docktermj 
I gave you the head and you want the tail ;)

The first error was:
LambdaFunctionGenerateCertificate
CREATE_FAILED
"Error occurred while GetObject. S3 Error Code: NoSuchBucket. S3 Error Message: The specified bucket does not exist (Service: AWSLambdaInternal; Status Code: 400; Error Code: InvalidParameterValueException; Request ID: 4ac00394-7b41-489f-9798-cecac57cc5e6; Proxy: null)"

I hope this helps.

**Comment by docktermj** [comment]
Hi @christianswanepoel !

Yep, now I know exactly what the issue is.  It's something we have to add for your AWS Region.  What AWS Region are you deploying your cloudformation into?

@antaenc We'll have to add a `senzing-public-xx-yyyy-n` S3 bucket for @christianswanepoel's AWS region.

**Comment by christianswanepoel** [comment]
I am running in eu-west-1

**Comment by docktermj** [comment]
@antaenc 

So we have an S3 bucket for `senzing-public-eu-west-2`.  Looks like we need an S3 bucket for `senzing-public-eu-west-1`.

@christianswanepoel In the meantime, you should be able to deploy into `senzing-public-eu-west-2` if that's an option for you.

**Comment by christianswanepoel** [comment]
Unfortunately I have a lot of code running in eu-west-1 interacting with Senzing (in VPC).

**Comment by docktermj** [comment]
@christianswanepoel That's OK.  We should be able to enable `eu-west-1` within a day or two (depending on @antaenc's schedule)

@antaenc I checked the AWS regions that don't support AWS Cognito.  They are: ARN(eu-north-1), CDG(eu-west-3), GRU(sa-east-1), SFO(us-west-1) and YUL(ca-central-1).   So the `eu-west-1` AWS Region should be OK.   Reference: https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services/

**Comment by docktermj** [comment]
@christianswanepoel The S3 bucket and its contents should now be available to you.  Give it a try.

**Comment by christianswanepoel** [comment]
So got further but stuck on:
LambdaRunnerCognitoCreateUser
CREATE_FAILED

Received response status [FAILED] from custom resource. Message returned: See the details in CloudWatch Log Stream: 2021/05/10/[$LATEST]8581f53c1b8a46abb0d484c43b663ecd (RequestId: 4ed111b0-3dcd-4e44-973e-f2933a1b75b6)

**Comment by docktermj** [comment]
@christianswanepoel,   Looking at the definition of the resource 

https://github.com/Senzing/aws-cloudformation-ecs-poc-simple/blob/804557c3350f9e4994da3a67f4bf24721f9b6d1b/cloudformation.yaml#L2212-L2222

I have a guess: Did you enter your email address at the beginning when prompted for user parameters?

To actually debug,  go to the cloudwatch logs (https://console.aws.amazon.com/cloudwatch/home?#logsV2:log-groups) and find the log further describing the error.

**Comment by christianswanepoel** [comment]
Ok let me try again - maybe I missed it

**Comment by docktermj** [comment]
Pull requests:

1. https://github.com/Senzing/aws-cloudformation-ecs-poc-simple/pull/78
